packages/theia-integration/src/browser/theia-model-source-changed-handler.ts

```diff
@@ -30,22 +30,23 @@ export class TheiaModelSourceChangedHandler extends ExternalModelSourceChangedHa
         if (element) {
             const widget = this.shell.findWidgetForElement(element);
             if (widget) {
-                const diagramWidget = this.getDiagramWidget(widget);
-                if (diagramWidget instanceof GLSPDiagramWidget) {
-                    return this.notifyModelSourceChangedWithWidget(diagramWidget, modelSourceName);
-                }
+                return this.notifyModelSourceChangedWithWidget(widget, modelSourceName);
             }
         }
         return [];
     }
 
-    protected async notifyModelSourceChangedWithWidget(diagramWidget: GLSPDiagramWidget, modelSourceName: string): Promise<Action[]> {
+    protected async notifyModelSourceChangedWithWidget(widget: Widget, modelSourceName: string): Promise<Action[]> {
+        const diagramWidget = this.getDiagramWidget(widget);
+        if (!(diagramWidget instanceof GLSPDiagramWidget)) {
```

Comment:
`this.getDiagramWidget()`  returns `GLSPDiagramWidget|undefined` so there is no need to typecheck here.
`if (!diagramWidget)` should be enough